### PR TITLE
test(e2e): add some e2e test cases

### PR DESCRIPTION
**Pull request by DavidChild**
**This PR includes   e2e test cases  as below**
1. net_peerCount
2. eth_feeHistory
3. mining
4. eth_subscribe
5. eth_syncing
6. net_peerCount
7. net_version
8. web3_clientVersion
9. eth_coinBase
10. eth_gasPrice
11. eth_getBalance
12. eth_getBlockByHash
13. eth_getBlockTransactionCountByHash
14. eth_getCode
15. eth_getStorageAt
16. eth_getTransactionByBlockHashAndIndex
17. eth_getTransactionByBlockNumberAndIndex
18. eth_getTransactionCount
19. eth_hasHrate
20. web3_sha3

**Other adjustments**
1.  some  code optimization

**No Breaking Change**

**Special notes for your reviewer:**

### CODE REVIEW

File: tests/e2e/src/eth_coinBase.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  </div>
+</body>
+<script>
+  document.getElementById("btn").addEventListener("click", () => {
+    const param1 = document.getElementById("param1").value;
+    const param2 =new Boolean(document.getElementById("param2").value);
+    const t_type = document.getElementById("testType").value;
+    const params = [];
+    const request_params = { method: "eth_coinbase", params: params }; // deafult is none params to request test.
+    if (t_type == "2") {
```
Comment (Hanssen0) [inline tests/e2e/src/eth_coinBase.html]: We prefer === in JS

File: tests/e2e/src/eth_coinBase.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  <div><input type="text" id="param1" value="" /></div>
+  <div><input type="text" id="param2" value="" /></div>
+  <div><input type="text" id="testType" value="" /></div>
+  <div> value: <span id="ret"></span></div>
+  </div>
+</body>
+<script>
+  document.getElementById("btn").addEventListener("click", () => {
+    const param1 = document.getElementById("param1").value;
+    const param2 =new Boolean(document.getElementById("param2").value);
```
Comment (Hanssen0) [inline tests/e2e/src/eth_coinBase.html]: A space was missed here

File: tests/e2e/src/create_test_data/createTestDataManage.js [hunk 3]
```diff
@@ -18,72 +18,71 @@ const transactionInfo = {
   transactionIndex: "",
   accountAddress: "",
 };
-
-const savejson = async (filePath, data) => {
-  const dataStr = JSON.stringify(data, null, 4);
-  if (dataStr) {
+const createTestDataMange = {
```
Comment (Hanssen0) [inline tests/e2e/src/create_test_data/createTestDataManage.js:21]: Maybe exporting these methods separately is better? I can't get this.

File: tests/e2e/src/eth_coinBase.test.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// eslint-disable-next-line
+import  goto from "./goto";
```
Comment (Hanssen0) [inline tests/e2e/src/eth_coinBase.test.js]: An extra space here

File: tests/e2e/src/eth_coinBase.test.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+const pageName = "eth_coinBase.html";
+describe("eth_coinBase", () => {
+  /**
+  * common real test
+  * axon haven't wallet now. so skip this test case
+  */
+  it("eth_coinBase_1", async () => {
+    await goto.goto(page, pageName);
```
Comment (Hanssen0) [inline tests/e2e/src/eth_coinBase.test.js]: Use `beforeAll` it will be faster.

File: tests/e2e/src/eth_getBalance.test.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
+import goto from "./goto";
+import testDataManage from "./create_test_data/createTestDataManage";
+
+let testDataInfo = null;
+const pageName = "eth_getBalance.html";
+describe("eth_getBalance", () => {
+  testDataInfo = testDataManage.readTestDataAsJson("testData_1.json");
```
Comment (Hanssen0) [inline tests/e2e/src/eth_getBalance.test.js:10]: use `beforeAll` to allow skipping these preparation if we don't need these test cases.

File: tests/e2e/src/eth_getBalance.test.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+  /**
+   * param1: real account address
+   * param2: real other number
+   */
+  it.skip("eth_getBalance_1", async () => {
+    await goto.goto(page, pageName);
+    const testType = await page.$(goto.pageIds.testTypeId);
+    const param1 = await page.$(goto.pageIds.param1Id);
+    const param2 = await page.$(goto.pageIds.param2Id);
```
Comment (Hanssen0) [inline tests/e2e/src/eth_getBalance.test.js:18]: Please use Promise.all to run these tasks asynchronously for performance.

**Review by Hanssen0 — CHANGES_REQUESTED** [review]
Please fix mentioned problems. Please improve those description of test cases following our contribution guide.

File: tests/e2e/src/create_test_data/createTestDataManage.js (hunk 3 above)
Comment (DavidChild) [reply tests/e2e/src/create_test_data/createTestDataManage.js:21]: This is more conducive to object encapsulation. In addition, the code is also more concise, and the place where this method is called is more clear about the method is belongs to which object.

Comment (DavidChild) [reply tests/e2e/src/create_test_data/createTestDataManage.js:21]: > Maybe exporting these methods separately is better? I can't get this.

This is more conducive to object encapsulation. In addition, the code is also more concise, and the place where this method is called is more clear about the method is belongs to which object.

File: tests/e2e/src/eth_coinBase.test.js (hunk 5 above)
Comment (DavidChild) [reply tests/e2e/src/eth_coinBase.test.js]: > Use `beforeAll` it will be faster.

ok!

File: tests/e2e/src/eth_getBalance.test.js (hunk 6 above)
Comment (DavidChild) [reply tests/e2e/src/eth_getBalance.test.js:10]: > use `beforeAll` to allow skipping these preparation if we don't need these test cases.

The assignment of each page is not affected, and each  case is isolated separately.

File: tests/e2e/src/eth_getBalance.test.js (hunk 7 above)
Comment (KaoImin) [reply tests/e2e/src/eth_getBalance.test.js:18]: Ref https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

**Review by Hanssen0 — APPROVED** [review]

File: tests/e2e/src/eth_getBalance.test.js (hunk 7 above)
Comment (KaoImin) [reply tests/e2e/src/eth_getBalance.test.js:18]: Why not change here？

Comment (DavidChild) [reply tests/e2e/src/eth_getBalance.test.js:18]: will be changes in anther pr. 

> Why not change here？

will be change in another pr.

**Pull request merged**